### PR TITLE
add Rubocop

### PR DESCRIPTION
**Pull request by jekyllbot**
PR automatically created for @DirtyF.

```text
add Rubocop
```

### CODE REVIEW

**Review by pathawks — APPROVED** [review]
Good call :+1:

File: Gemfile [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 
   gems = []
-  gems.concat config['gems'] if config.include? 'gems'
-  gems.concat config['plugins'] if config.include? 'plugins'
-  gems << 'pygments.rb' if config['highlighter'] == 'pygments'
-
+  gems.concat config["gems"] if config.include? "gems"
+  gems.concat config["plugins"] if config.include? "plugins"
+  gems << "pygments.rb" if config["highlighter"] == "pygments"
+  gems << "rubocop"
```
Comment (DirtyF) [inline Gemfile]: Is this the right way to add the gem. Travis is failing 😕

File: .rubocop.yml [hunk 2]
```diff
@@ -1,12 +1,7 @@
 inherit_gem:
   jekyll: .rubocop.yml
 
-Metrics/LineLength:
+AllCops:
+  TargetRubyVersion: 2.4
```
Comment (parkr) [inline .rubocop.yml]: This should be the lowest version we support.

File: .rubocop.yml [hunk 3]
```diff
@@ -1,12 +1,7 @@
...
   Exclude:
-    - jekyll-docs.gemspec
-
-Style/Documentation:
-  Enabled: false
-
-Style/FileName:
-  Enabled: false
+    - jekyll/**/*
```
Comment (parkr) [inline .rubocop.yml]: Exclude vendor/**/* too

File: Gemfile (hunk 1 above)
Comment (parkr) [reply Gemfile]: No, this block is only evaluated if jekyll is cloned. Add `gem “rubocop”` above `if Dir.exist`

**Review by parkr — CHANGES_REQUESTED** [review]

**Comment by DirtyF** [comment]
Build is passing now. Thanks for the review @parkr

**Comment by DirtyF** [comment]
@jekyllbot: merge +dev

**Pull request merged**